builder/azure/arm/step_snapshot_data_disks.go

```diff
@@ -0,0 +1,98 @@
+package arm
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2018-04-01/compute"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/hashicorp/packer/builder/azure/common/constants"
+	"github.com/hashicorp/packer/helper/multistep"
+	"github.com/hashicorp/packer/packer"
+)
+
+type StepSnapshotDataDisks struct {
+	client *AzureClient
+	create func(ctx context.Context, resourceGroupName string, srcUriVhd string, location string, tags map[string]*string, dstSnapshotName string) error
+	say    func(message string)
+	error  func(e error)
+}
+
+func NewStepSnapshotDataDisks(client *AzureClient, ui packer.Ui) *StepSnapshotDataDisks {
```

Comment:
Where are you planning to use this? I'm not seeing this function actually getting called anywhere.